openff/bespokefit/schema/fitting.py

```diff
@@ -81,6 +83,19 @@ class BaseOptimizationSchema(SchemaBase, abc.ABC):
         "stage.",
     )
 
+    @property
+    def target_smirks(self) -> List[SMIRKSType]:
+        """Returns a list of the target smirks types based on the selected hyper parameters.
+        Used to determine which parameters should be fit.
+        """
+        return list(
+            {
+                SMIRKSType(parameter.type)
+                for stage in self.stages
+                for parameter in stage.parameter_hyperparameters
+            }
+        )
+
     @property
     def initial_parameter_values(
         self,
```

Comment:
This maybe needs moving to the general schema as it will always be empty for the bespoke one?

Reply:
Yeah it's empty until we are inside the workflow. It could be useful if users want to compare the initial and final parameters in the fitting result?